data_management/process_bulk_uploads.py

```diff
@@ -69,86 +77,98 @@ def process_files(self, manifest_files_arr: list) -> list:
+                redcap_id = experiment["files"][0]["redcap_id"]
+                sample_id = experiment["files"][0]["spectrack_sample_id"]
+                if redcap_id and redcap_id.startswith("S-"):
+                    sample_id = redcap_id
+                    redcap_results = self.data_management.get_redcap_id_by_spectrack_sample_id(sample_id)
+                    if len(redcap_results) == 1:
+                        redcap_id = redcap_results[0]["spectrack_redcap_record_id"]
+                    else:
+                        redcap_id = ""
 
-                    if sample_id and len(self.data_management.get_participant_by_redcap_id(redcap_id)) > 0:
-                        files_copied = 0
-                        if "recruitment_site" in experiment:
-                            tis = experiment["recruitment_site"]
+                if not sample_id:
+                    sample_id = redcap_id
+
+                if sample_id and len(self.data_management.get_participant_by_redcap_id(redcap_id)) > 0:
+                    if "recruitment_site" in experiment:
+                        tis = experiment["recruitment_site"]
+                    else:
+                        tis = ""
+                    logger.info(f"Trying to add package for {redcap_id}")
+                    dlu_file_list = self.process_files(experiment["files"])
+                    if package_type == PackageType.SEGMENTATION:
+                        dlu_file_list.append(self.get_single_file(SEGMENTATION_README))
+                        tis = "UFL"
+                    result = self.data_management.dlu_mongo.find_by_package_type_and_redcap_id(package_type.value, sample_id)
+                    if result is None:
+                        logger.info(f"Adding package for {redcap_id}")
+                        package = DLUPackage()
+                        package.dlu_package_type = package_type.value
+                        package.dlu_tis = tis
+                        package.dlu_created = datetime.datetime.today()
+                        package.dlu_submitter = DBRef(collection='users', id=ObjectId(self.submitter))
+                        package.submitter_name = self.submitter_name
+                        package.dlu_data_generators = manifest_data["data_generators"]
+                        package.dlu_protocol = ""
+                        package.dlu_description = manifest_data["dataset_description"]
+                        package.dlu_tis_experiment_id = experiment["internal_experiment_id"]
+                        package.dlu_subject_id = sample_id
+                        package.dlu_lfu = True
+                        package.redcap_id = redcap_id
+                        package.known_specimen = sample_id
+                        package.dlu_version = 4
+                        package.dlu_dataset_information_version = 1
+                        if self.globus_only:
+                            package.globus_dlu_status = 'waiting'

```

Comment:
Is this correct? Doesn't waiting mean that Rachel has set them as ready to move into the data lake proper? I thought this should not be set at this point, but I could be misremembering.